projects/i18n/src/modules/i18n/get-lib-string-for-locale.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  SkyLibResources
+} from './lib-resources';
+
+/**
+ * @internal
+ */
+ export function getLibStringForLocale(
+  resources: {
+    [locale: string]: SkyLibResources
+  },
+  preferredLocale: string,
+  name: string
+): string {
+  const defaultLocale = 'en-US';
+
+  function getResourcesForLocale(locale: string): SkyLibResources {
+    const parsedLocale = locale.toUpperCase().replace('_', '-');
+    return resources[parsedLocale];
+  }
+
+  let values: SkyLibResources = getResourcesForLocale(preferredLocale);
+
+  if (values && values[name]) {
+    return values[name].message;
+  }
+
+  // Attempt to locate default resources.
+  values = getResourcesForLocale(defaultLocale);
+
+  if (values && values[name]) {
+    return values[name].message;
+  }
+
+  return name;
```

Comment:
`getStringForLocale` just returned an empty string if no match was found. Was this change to returning the `name` deliberate?

Reply:
Yes. I wanted to mimic what "app" resources service is doing today: https://github.com/blackbaud/skyux-i18n/blob/master/src/app/public/modules/i18n/resources.service.ts#L208